lib/train/transports/vmware.rb

```diff
@@ -0,0 +1,190 @@
+# encoding: utf-8
+require 'train/plugins'
+require 'open3'
+require 'ostruct'
+require 'json'
+require 'mkmf'
+
+module Train::Transports
+  class VMware < Train.plugin(1)
+    name 'vmware'
+    option :viserver, default: ENV['VISERVER']
+    option :username, default: ENV['VISERVER_USERNAME']
+    option :password, default: ENV['VISERVER_PASSWORD']
+    option :insecure, default: false
+
+    def connection(_ = nil)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection # rubocop:disable ClassLength
+      POWERSHELL_PROMPT_REGEX = /PS\s.*> $/
+
+      def initialize(options)
+        super(options)
+
+        options[:viserver] = options[:viserver] || options[:host]
+        options[:username] = options[:username] || options[:user]
+
+        @username = options[:username]
+        @viserver = options[:viserver]
+        @session = nil
+        @stdout_buffer = ''
+        @stderr_buffer = ''
+
+        @powershell_binary = detect_powershell_binary
+
+        if @powershell_binary == :powershell
+          require 'train/transports/local'
+          @powershell = Train::Transports::Local::Connection.new(options)
+        end
+
+        if options[:insecure] == true
+          run_command_via_connection('Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Scope Session -Confirm:$False')
+        end
+
+        @platform_details = {
+          release: "vmware-powercli-#{powercli_version}",
+        }
+
+        connect
+      end
+
+      def connect
+        login_command = "Connect-VIServer #{options[:viserver]} -User #{options[:username]} -Password #{options[:password]} | Out-Null"
+        result = run_command_via_connection(login_command)
+
+        if result.exit_status != 0
+          message = "Unable to connect to VIServer at #{options[:viserver]}. "
+          case result.stderr
+          when /Invalid server certificate/
+            message += 'Certification verification failed. Please use `--insecure` or set `Set-PowerCLIConfiguration -InvalidCertificateAction Ignore` in PowerShell'
+          when /incorrect user name or password/
+            message += 'Incorrect username or password'
+          else
+            message += result.stderr.gsub(/-Password .*\s/, '-Password REDACTED')
+          end
+
+          raise message
+        end
+      end
+
+      def local?
+        true
```

Comment:
Is this constant?

Reply:
I think so, I'm trying to override: https://github.com/inspec/train/blob/d7b958c4cd1ee3b7993bd0fc05a85fb25544046c/lib/train/plugins/base_connection.rb#L76